taxii2client/v20/__init__.py

```diff
@@ -34,7 +34,7 @@ def as_pages(func, start=0, per_request=0, *args, **kwargs):
     yield _to_json(resp)
     total_obtained, total_available = _grab_total_items(resp)
 
-    if total_obtained != per_request:
+    if total_available > per_request and total_obtained != per_request:
         log.warning("TAXII Server response with different amount of objects! Setting per_request=%s", total_obtained)
         per_request = total_obtained
```

Comment:
The observation I have is that at least on the first request you can "fix" the amount of objects per request based on the first one (look at line 39: `per_request = total_obtained`). Changing the logic to what you suggest would prevent that from happening, and would have unintended behavior for subsequent requests.

Are you more concerned about the message being very noisy (triggers a lot)? Or the fact it is a warning (maybe we could lower it to info or debugging)? I am trying to better understand the reasoning behind the proposed change.

Reply:
If the server returns all of the results on the first request and that is less than per_request no subsequent requests will be made, so there is no need to warn the user the server returned less than was asked for.

For example, if there are ten objects on the server, and you ask for them 50 at a time, you will get all of them in the first request. In this case issuing a warning makes it seem like the server did something unexpected.